webrtc-stats.html

```diff
@@ -3622,6 +3632,107 @@ <h2>
           </section>
         </div>
       </section>
+      <section id="stunserverconnection-dict*">
+        <h3>
+          <dfn>RTCStunServerConnectionStats</dfn> dictionary
+        </h3>
+        <div>
+          <pre class="idl">dictionary RTCStunServerConnectionStats : RTCStats {
+             DOMString url;
+             DOMString address;
+             long port;
+             DOMString protocol;
+             RTCNetworkType networkType;
+             unsigned long totalRequestsSent;
+             unsigned long totalResponsesReceived;
+             double totalRoundTripTime;
+  };</pre>
+          <section>
+            <h2>
+              Dictionary <a class="idlType">RTCStunServerConnectionStats</a> Members
+            </h2>
+            <dl data-link-for="RTCStunServerConnectionStats" data-dfn-for=
+            "RTCStunServerConnectionStats" class="dictionary-members">
+              <dt>
+                <dfn><code>url</code></dfn> of type <span class=
+                "idlMemberType"><a>DOMString</a></span>
+              </dt>
+              <dd>
+                <p>
+                  The URL of the TURN or STUN server.
+                </p>
+              </dd>
+              <dt>
+                <dfn><code>address</code></dfn> of type <span class=
+                "idlMemberType"><a>DOMString</a></span>
+              </dt>
+              <dd>
+                <p>
+                  It is the IP address of the local interface, allowing for IPv4 addresses
```

Comment:
Just from the name, one would expect "address" to be the address of the STUN server, but this is a local interface address (sort of). Section 15.1 of RFC 5245 contains two addresses (connection-address and rel-addr), so isn't a precise enough reference.

If it's rel-addr (and rel-port, below), please use those names, suitably camel-cased. Note also that this may be different from the browser's address - see 5245 appendix B.3, which says that "the server reflexive address towards the TURN server is needed".

Revealing rel-addr is also a privacy issue - see JSEP section 3.5.3. This needs mentioning, and what to do if it's not exposed (I would recommend not returning the member rather than returning 0.0.0.0).

Reply:
i looked into this a bit, and rel-addr seems not the right fit for this.
the problem is that rel-addr (afaik) relies on a response from the STUN/TURN server. this would mean that a response is required to identify the connection. however, in case of a failed connection to this server, it is valuable information to know which local interface is responsible.
so i'm wondering if it would be possible to actually get the local interface address, i.e. something that identifies the origin of the connection independent of a response?

Reply:
why not use the URL of the STUN/TURN server? That's what's configured.